src/utils/manager.py

```diff
@@ -197,15 +213,44 @@ def mount(
         _logger.debug(f"Creating mountpoint {target}")
         target.mkdir()
 
-    cmd = ["mount", "-t", "nfs"]
+    autofs_id = _mountpoint_to_autofs_id(target)
+
+    mount_opts = ""
     if options:
         if port:
             options.append(f"port={port}")
-        cmd.extend(["-o", ",".join(options)])
-    cmd.extend([endpoint, target])
+        mount_opts = ",".join(options)
+
+    _logger.debug(f"Mounting NFS share endpoint {endpoint} at {target} with options {options}")
+
+    if mount_opts:
+        master = f"/- /etc/auto.{autofs_id} {mount_opts}"
+    else:
+        master = f"/- /etc/auto.{autofs_id}"
+
+    mount_data = [
+        FileData(
+            path=f"/etc/auto.master.d/{autofs_id}.autofs",
+            contents=master,
+        ),
+        FileData(
+            path=f"/etc/auto.{autofs_id}",
+            contents=f"{target} {endpoint}",
+        ),
+    ]
+
+    for filedata in mount_data:
+        try:
+            with open(filedata.path, "w") as f:
+                f.write(filedata.contents)
+        except OSError as e:
+            _logger.debug(f"Failed to create file `{filedata.path}`. Reason:\n{e}")
+            raise Error(f"failed to create file `{filedata.path}`")
```

Comment:
I don't think we need the for loop here. I would only advocate the for loop here if we were to have a variable amount of files that we need to operator on, but it looks here that we only need to operate on two files. We could likely simplify the implementation to something like the following:

```python3
mount_opts = ""
if options:
    if port:
         options.append(f"port={port}")
    mount_opts = ",".join(options)

pathlib.Path(f"/etc/auto.master.d/{autofs_id}.autofs").write_text(f"/- /etc/autofs.{autofs_id}" + f" {mount_opts}" if mount_opts else "")
pathlib.Path(f"/etc/auto.{autofs_id}").write_text(f"{target} {endpoint}")
```

Reply:
Will apply. Though, I think `write_test` throws if the file doesn't exist, so I'll have to do it in a slightly different way.

Reply:
I think you mean `write_text` :smile: 

Nope - you should be good. `write_text(...)` will create the file if it does not exist. The only thing to note is that it will overwrite the contents of the file if it already exists. You'd only only want to use the `.touch(...)` method of the `Path(...)` object if you wanted to create the file with custom permissions.